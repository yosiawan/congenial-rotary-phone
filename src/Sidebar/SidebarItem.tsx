export type TSidebarItemProps = {
  onClick: () => void;
  isActive: boolean;
  icon: any;
  title: string;
};

export default function SidebarItem(props: TSidebarItemProps) {
  return (
    <div
      id="sidebar-item"
      onClick={props.onClick}
      className={`main-sidebar__item ${props.isActive ? "active" : "inactive"}`}
    >
      <div>{props.icon}</div>
      <div>{props.title}</div>
    </div>
  );
}
