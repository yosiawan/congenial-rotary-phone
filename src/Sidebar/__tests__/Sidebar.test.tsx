// export default null;
import { shallow, mount } from "enzyme";
import SidebarItem, { TSidebarItemProps } from "../SidebarItem";
import HomeIcon from "@mui/icons-material/Home";

describe("Sidebar Item test", () => {
  let masterProps: TSidebarItemProps = {
    onClick: jest.fn(),
    isActive: true,
    icon: <HomeIcon />,
    title: "DMMY_TITLE",
  };

  it("Renders successfully", () => {
    expect(shallow(<SidebarItem {...masterProps} />)).toMatchSnapshot();
  });

  it("Renders title correctly", () => {
    const render = mount(<SidebarItem {...masterProps} />);
    expect(render.html().includes("DMMY_TITLE")).toBeTruthy();
  });

  it("Has 'active' class when prop isActive is 'true'", () => {
    const render = shallow(<SidebarItem {...masterProps} />);
    expect(render.exists(".active")).toBeTruthy();
  });

  it("Does not have 'active' class when prop isActive is 'false'", () => {
    const props: TSidebarItemProps = {
      ...masterProps,
      isActive: false,
    };
    const render = shallow(<SidebarItem {...props} />);
    expect(render.exists(".active")).toBeFalsy();
  });

  it("Click handler is called on user click", () => {
    const mockOnClickHandler = jest.fn();
    const props: TSidebarItemProps = {
      ...masterProps,
      onClick: mockOnClickHandler,
    };
    const render = shallow(<SidebarItem {...props} />);
    render.find("#sidebar-item").simulate("click");
    expect(mockOnClickHandler).toHaveBeenCalledTimes(1);
  });
});
