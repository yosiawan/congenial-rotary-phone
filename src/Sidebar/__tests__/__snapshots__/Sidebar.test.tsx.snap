// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar View test Renders successfully 1`] = `ShallowWrapper {}`;

exports[`Sidebar View test renders successfully 1`] = `ShallowWrapper {}`;
